src/modules/Channel/components/SuggestedMentionList/index.tsx

```diff
@@ -104,32 +114,51 @@ function SuggestedMentionList(props: SuggestedMentionListProps): JSX.Element {
-      nicknameStartsWithFilter: searchString.slice(USER_MENTION_TEMP_CHAR.length),
-    });
     // Add member list query for customization
-    query.next()
-      .then((memberList) => {
-        const suggestingMembers = memberList
-          .filter((member) => currentUserId !== member?.userId)
-          .slice(0, maxSuggestionCount);
-        if (suggestingMembers.length < 1) {
-          logger.info('SuggestedMentionList: Fetched member list is empty');
-        } else {
-          logger.info('SuggestedMentionList: Fetching member list succeeded', { memberListQuery: query, memberList: suggestingMembers });
-          setCurrentUser(suggestingMembers[0]);
-        }
-        setLastSearchString(searchString);
-        onFetchUsers(suggestingMembers);
-        setCurrentMemberList(suggestingMembers);
-      })
-      .catch((error) => {
-        if (error) {
-          logger.error('SuggestedMentionList: Fetching member list failed', error);
-        }
+    if (currentChannel.isSuper) {
+      const query = channelInstance?.createMemberListQuery({
+        limit: maxSuggestionCount + 1, // because current user could be included
+        nicknameStartsWithFilter: searchString.slice(USER_MENTION_TEMP_CHAR.length),
       });
-  }, [channelInstance?.url, searchString]);
+      query.next()
+        .then((memberList) => {
+          const suggestingMembers = memberList
+            .filter((member) => currentUserId !== member?.userId)
+            .slice(0, maxSuggestionCount);
+          if (suggestingMembers.length < 1) {
+            logger.info('SuggestedMentionList: Fetched member list is empty');
+          } else {
+            logger.info('SuggestedMentionList: Fetching member list succeeded', { memberListQuery: query, memberList: suggestingMembers });
+            setCurrentUser(suggestingMembers[0]);
+          }
+          setLastSearchString(searchString);
+          onFetchUsers(suggestingMembers);
+          setCurrentMemberList(suggestingMembers);
+        })
+        .catch((error) => {
+          if (error) {
+            logger.error('SuggestedMentionList: Fetching member list failed', error);
+          }
+        });
+    } else {
+      const suggestingMembers = getCurrentSearchedMemberList();
+      if (suggestingMembers.length < 1) {
+        logger.info('SuggestedMentionList: Channel\'s member list is empty');
+      } else {
+        logger.info('SuggestedMentionList: Channel\'s member list has been updated.', { memberList: suggestingMembers });
+        setCurrentUser(suggestingMembers[0]);
+      }
+      setLastSearchString(searchString);
+      onFetchUsers(suggestingMembers);
+      setCurrentMemberList(suggestingMembers);
+    }
+  }, [
+    channelInstance?.url,
+    channelInstance.members.length,
+    // We have to be specific like this or React would not recognize the changes in instances.
+    channelInstance.members.map((member: Member) => member.state).join(),
+    searchString,
+  ]);
```

Comment:
I think we can divide this useEffect into two
one is using the query, and the one is using members

Reply:
I believe it's not a good idea to separate the same business logic because it would require managing two useEffect blocks.
Simply by determining where to fetch the user based on isSuper, you can decide where to handle user retrieval. Instead of separating the useEffect itself, it would be more beneficial to split the function that retrieves members into two, reducing the complexity of the useEffect and making the code more readable.